# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1729444259 -7200
# Parent  bfcd63e4b2a0bf1e8e18697d456e66df16eac305
Bug 1924356 - Make sure Total Message Count is displayed for Favorite Folders.

Bonus fix: For virtual folders, sizeOnDisk() fails.

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -4398,8 +4398,9 @@ class FolderTreeRow extends HTMLLIElemen
 
   set totalCount(value) {
     this.classList.toggle("total", value > 0);
     this.totalCountLabel.textContent = value;
+    this.totalCountLabel.hidden = !folderPane.isTotalMsgCountVisible();
     this.#updateAriaLabel();
   }
 
   /**
@@ -4412,8 +4413,11 @@ class FolderTreeRow extends HTMLLIElemen
   }
 
   set folderSize(value) {
     this.folderSizeLabel.textContent = value;
+    this.folderSizeLabel.hidden = !folderPane.isItemVisible(
+      "folderPaneFolderSize"
+    );
     this.#updateAriaLabel();
   }
 
   #updateAriaLabel() {
diff --git a/mailnews/local/src/nsLocalMailFolder.cpp b/mailnews/local/src/nsLocalMailFolder.cpp
--- a/mailnews/local/src/nsLocalMailFolder.cpp
+++ b/mailnews/local/src/nsLocalMailFolder.cpp
@@ -953,8 +953,13 @@ NS_IMETHODIMP nsMsgLocalMailFolder::GetS
   nsresult rv = GetIsServer(&isServer);
   // If this is the rootFolder, return 0 as a safe value.
   if (NS_FAILED(rv) || isServer) mFolderSize = 0;
 
+  // Ignore virtual folders, for maildir there's not even a file to test.
+  uint32_t folderFlags;
+  GetFlags(&folderFlags);
+  if (folderFlags & nsMsgFolderFlags::Virtual) mFolderSize = 0;
+
   if (mFolderSize == kSizeUnknown) {
     nsCOMPtr<nsIFile> file;
     rv = GetFilePath(getter_AddRefs(file));
     NS_ENSURE_SUCCESS(rv, rv);
